File: java/ABTestExample/app/src/main/java/com/pubscale/sdkone/example/abtestexample/SplashActivity.java
package com.pubscale.sdkone.example.abtestexample;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.pubscale.sdkone.example.abtestexample.ad_formats.AppOpen;
import com.pubscale.sdkone.example.abtestexample.databinding.ActivitySplashBinding;
import com.pubscale.sdkone.example.abtestexample.event_listener.AppOpenAdEventListener;

public class SplashActivity extends AppCompatActivity {

    private ActivitySplashBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySplashBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        loadAppOpenAd();
    }

    private void loadAppOpenAd() {
        AppOpen appOpenAd = AppOpen.getInstance();
        AppOpenAdEventListener appOpenAdEventListener = new AppOpenAdEventListener() {
            @Override
            public void onAdLoading() {
                binding.appOpenStatus.setText("Loading...");
            }

            @Override
            public void onAdLoaded() {
                binding.appOpenStatus.setText("Loaded");
                appOpenAd.showAd(SplashActivity.this);
            }

            @Override
            public void onAdLoadFailed() {
                binding.appOpenStatus.setText("Failed");
                openMainActivity();
            }

            @Override
            public void onAdShowFailed() {
                binding.appOpenStatus.setText("Show failed");
                openMainActivity();
            }

            @Override
            public void onAdOpened() {
                binding.appOpenStatus.setText("Opened");
            }

            @Override
            public void onAdClosed() {
                binding.appOpenStatus.setText("Closed");
                openMainActivity();
            }

            @Override
            public void onAdDisabled() {
                binding.appOpenStatus.setText("Disabled");
                Toast.makeText(SplashActivity.this, "App open ad disabled", Toast.LENGTH_SHORT).show();
                openMainActivity();
            }
        };

        appOpenAd.setAppOpenAdEventListener(appOpenAdEventListener);
        appOpenAd.loadAd(this);
    }

    private void openMainActivity() {
        startActivity(new Intent(SplashActivity.this, MainActivity.class)
                .setFlags((Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK)));
    }
}

File: kotlin/ABTestExample/app/src/main/java/com/pubscale/sdkone/example/abtestexample/SplashActivity.kt
package com.pubscale.sdkone.example.abtestexample

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.pubscale.sdkone.example.abtestexample.ad_formats.AppOpen
import com.pubscale.sdkone.example.abtestexample.databinding.ActivitySplashBinding
import com.pubscale.sdkone.example.abtestexample.event_listener.AppOpenAdEventListener

class SplashActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySplashBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivitySplashBinding.inflate(layoutInflater)
        setContentView(binding.root)

        loadAppOpenAd()
    }

    private fun loadAppOpenAd() {
        val appOpenAdEventListener: AppOpenAdEventListener = object : AppOpenAdEventListener {
            override fun onAdLoading() {
                binding.appOpenStatus.text = "Loading..."
            }

            override fun onAdLoaded() {
                binding.appOpenStatus.text = "Loaded"
                AppOpen.showAd(this@SplashActivity)
            }

            override fun onAdLoadFailed() {
                binding.appOpenStatus.text = "Failed"
                openMainActivity()
            }

            override fun onAdShowFailed() {
                binding.appOpenStatus.text = "Show failed"
                openMainActivity()
            }

            override fun onAdOpened() {
                binding.appOpenStatus.text = "Opened"
            }

            override fun onAdClosed() {
                binding.appOpenStatus.text = "Closed"
                openMainActivity()
            }

            override fun onAdDisabled() {
                binding.appOpenStatus.text = "Disabled"
                Toast.makeText(this@SplashActivity, "App open ad disabled", Toast.LENGTH_SHORT)
                    .show()
                openMainActivity()
            }
        }
        AppOpen.setAppOpenAdEventListener(appOpenAdEventListener).loadAd(this)
    }

    private fun openMainActivity() {
        startActivity(
            Intent(this@SplashActivity, MainActivity::class.java)
                .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        )
    }
}